src/calling/haplotype_abundances.rs

```diff
@@ -29,18 +54,37 @@ impl Caller {
         for i in 0..num_bootstraps {
             let dataset = hdf5.dataset(&format!("bootstrap/bs{i}", i = i))?;
             let tpm = dataset.read_1d::<u64>()?;
-            let tst = tpm[index];
+            let tst = tpm[index]; //a better variable name
             bootstraps.push(tst);
         }
-        let ecdf = kernel_density::ecdf::Ecdf::new(&bootstraps);
+        let mut ecdf = kernel_density::ecdf::Ecdf::new(&bootstraps);
+        let ecdf: Vec<f64> = bootstraps.iter().map(|&x| ecdf.value(x)).collect();
         Ok(ECDF {
             seqname,
             bootstraps,
             ecdf,
         })
+    
     }
 
     pub(crate) fn call(&self) {
         todo!()
     }
 }
+
+impl ECDF {
+    pub(crate) fn qc(&self) -> Result<()> {
+        let blueprint = "../../templates/plots/qc_cdf.json";
+        let mut blueprint: serde_json::Value = serde_json::from_str(blueprint)?;
+
+        let bootstraps = &self.bootstraps;
+        let ecdf = &self.ecdf;
+
+        let data = json!([bootstraps, ecdf]);
+        blueprint["data"]["values"] = data;
+
+        println!("{}", serde_json::to_string_pretty(&blueprint)?);
```

Comment:
There should be a separate JSON for each sequence. Hence, we cannot print them here, but they should be stored inside a given directory.